homeassistant/components/recorder/core.py

```diff
@@ -878,7 +877,7 @@ def _process_state_changed_event_into_session(self, event: Event) -> None:
             dbstate = States.from_event(event)
             shared_attrs_bytes = StateAttributes.shared_attrs_bytes_from_event(
                 event,
-                self._entity_registry,
+                entity_sources(self.hass),
```

Comment:
We could keep a reference to the `entity_sources` dict in the recorder instance if it's initialized when starting Home Assistant, in the same way as we load the registries early.